### PR TITLE
Sentry: Tag production deployment status

### LINKED ISSUE #1942: Error: GraphQL query failed to execute: Bad Request
Sentry Issue: [REACT-COMMERCE-3EF](https://ifixit.sentry.io/issues/4421858774/?referrer=github_integration)

```
Error: GraphQL query failed to execute: Bad Request
  File "app:///_next/server/chunks/478.js", line 3890, in requester
    throw new Error(`GraphQL query failed to execute: ${response.statusText}`);
  File "app:///_next/server/chunks/478.js", line 4028, in findProductList
    const result = await strapi_sdk/* strapi.getProductList */.s1.getProductList({
  File "app:///_next/server/pages/Tools.js", line 54, in getServerSideProps
    const [globalSettings, stores, currentStore, productList] = await Promise.all([(0,_models_global_settings__WEBPACK_IMPORTED_MODULE_4__/* . {snip}
...
(4 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by masonmcelvain**
Closes #1942

Adds a boolean tag (`current_production`) to all Sentry events indicating whether the deployment is the current production deployment.

## QA

Throw an error somewhere in local dev. Make sure the new tag (`current_production`) shows up in Sentry. It should be `false` for local dev.

### CODE REVIEW

**Review by jarstelfox — DISMISSED** [review]
CR 🌵 but deploy_block 🌵 I think it would be worth while to add the sha as a release as we do in the PHP app: 

https://github.com/iFixit/ifixit/blob/596fe7a6879b33d41ef0c82eb609a53561064b89/iFixit/Framework/Errors/SentryReporter.php#L74
https://github.com/iFixit/ifixit/blob/596fe7a6879b33d41ef0c82eb609a53561064b89/iFixit/Framework/Errors/SentryReporter.php#L263-L273

**Comment by masonmcelvain** [comment]
>  I think it would be worth while to add the sha as a release as we do in the PHP app

I omitted it here because it seems that Sentry is automatically detecting the sha and assigning it to a release tag.

That said, I'm happy to do it explicitly. dev_block ✌🏻

**Comment by jarstelfox** [comment]
>it seems that Sentry is automatically detecting the sha and assigning it to a release tag.

I'd leave it and un_deploy_block 🌵.

The main reason is assigning commits to a release is an extra step or two that requires a GH action on deploy. If it just works let's keep it a non-issue 👍🏻

**Comment by masonmcelvain** [comment]
Ok! un_dev_block ✌🏻 

I'd also like your opinion on whether a tag is suitable here, or if we should drop the event entirely (i.e. don't report to Sentry if it's a non-current production deployment). I thought that was a bit extreme, and the tag empowers us to filter in the dashboard at least.

**Comment by masonmcelvain** [comment]
dev_block ✌🏻 @erinemay we're going to take a different approach

https://ifixit.slack.com/archives/C01GMQXJC95/p1693937453575549

**Comment by danielbeardsley** [comment]
I feel pretty weird having our code curl github before reporting every error.

Then again, I'm not sure of another way to get this done. Perhaps we limit each page load to reporting a single error? Perhaps we cache the "current production" value for a few minutes?

**Comment by masonmcelvain** [comment]
> I feel pretty weird having our code curl github before reporting every error.
> 
> Then again, I'm not sure of another way to get this done. Perhaps we limit each page load to reporting a single error? Perhaps we cache the "current production" value for a few minutes?

It is weird, but I can't think of an alternative solution with the [system flags](https://vercel.com/docs/projects/environment-variables/system-environment-variables) Vercel exposes to us.

I tried implementing the caching solution you suggested.

**Comment by masonmcelvain** [comment]
un_dev_block ✌🏻 We ended up deciding this solution is good enough for now (though not ideal). Maybe we'll change the Sentry release to a datetime or timestamp in another PR.

File: frontend/sentry.client.config.ts [hunk 1]
```diff
@@ -2,12 +2,25 @@
...
 Sentry.init({
+   async beforeSend(event) {
+      try {
+         const current_production = await isCurrentProductionDeployment();
+         event.tags = { ...event.tags, current_production };
+      } catch (e) {
+         event.extra = {
+            ...event.extra,
+            'Exception Checking Production Deployment': e,
+         };
```
Comment (jarstelfox) [inline frontend/sentry.client.config.ts:20]: Minor: I would also add a new tag indicating that the `event.extra.'Exception Checking Production Deployment'` exists.

Something like: `SecondaryError: true`

Just makes it easy to search

Comment (masonmcelvain) [reply frontend/sentry.client.config.ts:21]: Good idea! https://github.com/iFixit/react-commerce/pull/1958/commits/053774c826fb44d90ffbcad910581ee4c79d346a

**Review by jarstelfox — APPROVED** [review]
CR 🌵

**Comment by erinemay** [comment]
QA 👍 I caused an error and saw the new tag present: 
image

It was false for local_dev

**Comment by masonmcelvain** [comment]
We got a `before_send_error` on this [event](https://ifixit.sentry.io/share/issue/6b81069c611d47dc913c0b30b1fc3141/), but the exception we logged was an empty object. Not sure what to make of it 🤷🏻 

image

**Pull request merged**